src/NuGet.Core/NuGet.LibraryModel/DownloadDependency.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using NuGet.Shared;
+using NuGet.Versioning;
+
+namespace NuGet.LibraryModel
+{
+    public class DownloadDependency : IEquatable<DownloadDependency>, IComparable<DownloadDependency>
+    {
+        public string Name { get; }
+
+        public VersionRange VersionRange { get; }
+
+        public DownloadDependency(
+            string name,
+            VersionRange versionRange)
```

Comment:
could there be auto reference vs explicit download dependency now or in future? if yes, should there be some provision to differentiate for future reference?